File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        res = 0
        for i in range(len(prices)):
            for j in range(i+1,len(prices)):
                if prices[i] > prices[j]:
                    continue
                res = max(prices[j] - prices[i], res)
        return res

File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-0.cpp
class Solution {
public:
    int characterReplacement(string s, int k) {
        unordered_set<char> charSet(s.begin(), s.end());
        int res = 0;

        for (char c: charSet) {
            int count = 0, l = 0;
            for (int r=0; r<s.size(); r++) {
                if (s[r] == c) count++;
                while ((r-l+1)-count > k) { // the window needs >k replacements
                    if (s[l] == c) count--;
                    l++;                    // move l
                }
                res = max(res, r-l+1);
            }
        }
        return res;
    }
};
